File: semgus-sketch/src/main/java/org/semgus/sketch/base/Syntax.java
package org.semgus.sketch.base;

import org.jetbrains.annotations.NotNull;
import org.semgus.pretty.Code;
import org.semgus.pretty.Prettiable;
import org.semgus.sketch.frontend.Distiller;

/**
 * Sketch syntax.
 */
public sealed interface Syntax extends Prettiable permits Expr, Stmt, Id, Param {
  @Override default @NotNull Code codify() {
    return Distiller.distill(this);
  }
}

File: semgus-sketch/src/main/java/org/semgus/sketch/base/syntax/Syntax.kt
package org.semgus.sketch.base.syntax

import org.semgus.pretty.Prettiable
import org.semgus.sketch.codegen.distill

/**
 * Sketch syntax.
 */
internal sealed class Syntax : Prettiable {
  override fun codify() = distill()
}
